Sources/ContainerClient/Core/Volume.swift

```diff
@@ -0,0 +1,223 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// A named volume that can be mounted in containers.
+public struct Volume: Sendable, Codable, Equatable {
+    public var name: String
+    // Driver used to create the volume.
+    public var driver: String
+    // The mount point of the volume on the host.
+    public var mountpoint: String
+    public var createdAt: Date
+    // User-defined key/value metadata.
+    public var labels: [String: String]
+    // Driver-specific options.
+    public var options: [String: String]
+    public var scope: VolumeScope
+    public var size: UInt64?
+    public var usageData: VolumeUsageData?
+
+    public init(
+        name: String,
+        driver: String = "local",
+        mountpoint: String,
+        createdAt: Date = Date(),
+        labels: [String: String] = [:],
+        options: [String: String] = [:],
+        scope: VolumeScope = .local,
+        size: UInt64? = nil,
+        usageData: VolumeUsageData? = nil
+    ) {
+        self.name = name
+        self.driver = driver
+        self.mountpoint = mountpoint
+        self.createdAt = createdAt
+        self.labels = labels
+        self.options = options
+        self.scope = scope
+        self.size = size
+        self.usageData = usageData
+    }
+}
+
+/// Scope of volume availability.
+public enum VolumeScope: String, Sendable, Codable, CaseIterable {
+    case local = "local"
+    case global = "global"
+}
+
+/// Usage information for a volume.
+public struct VolumeUsageData: Sendable, Codable, Equatable {
+    public var size: Int64
+    // Number of containers currently using this volume.
+    public var refCount: Int64
+
+    public init(size: Int64, refCount: Int64) {
+        self.size = size
+        self.refCount = refCount
+    }
+}
+
+/// Request to create a new volume.
+public struct VolumeCreateRequest: Sendable, Codable {
+    public var name: String
+    public var driver: String
+    public var driverOpts: [String: String]
+    public var labels: [String: String]
+
+    public init(
+        name: String,
+        driver: String = "local",
+        driverOpts: [String: String] = [:],
+        labels: [String: String] = [:]
+    ) {
+        self.name = name
+        self.driver = driver
+        self.driverOpts = driverOpts
+        self.labels = labels
+    }
+}
+
+/// Request to delete a volume.
+public struct VolumeDeleteRequest: Sendable, Codable {
+    public var name: String
+    public var force: Bool
+
+    public init(name: String, force: Bool = false) {
+        self.name = name
+        self.force = force
+    }
+}
+
+/// Response containing a list of volumes.
+public struct VolumeListResponse: Sendable, Codable {
+    public var volumes: [Volume]
+    public var warnings: [String]?
```

Comment:
Why `[Type]?` and not just `[Type]`? Optional-of-list is often redundant and needlessly complicates handling.